pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -106,60 +109,75 @@ func (m topologyToMatchedTermCount) clone() topologyToMatchedTermCount {
 	return copy
 }
 
+func (m topologyToMatchedTermCount) updateWithTerm(node *v1.Node, t *framework.AffinityTerm, value int64) {
+	if tv, ok := node.Labels[t.TopologyKey]; ok {
```

Comment:
Can we do this check before matching the terms, then we save those expensive term matching. But this can be done in another PR since this is not a thing introduced in this PR.